### PR TITLE
Add name mangling for second constructors and function overloading

### PR DESCRIPTION
**Pull request by gabryon99**
When a ConstructorSymbol's name is embedded, its type signature will be used to generate the name. Non-nullable types will be prefixed with a `T`. Nullable types with `NT`.

### CODE REVIEW

File: plugins/formal-verification/testData/diagnostics/no_contracts/class_constructors.fir.txt
Comment (22slin22) [inline plugins/formal-verification/testData/diagnostics/no_contracts/class_constructors.fir.txt:1]: FIR files should not be created anymore, if you have already rebased onto the newest changes could you delete these files manually.

File: plugins/formal-verification/testData/diagnostics/no_contracts/class_constructors.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    constructor(x1: Int, x2: Int) {
+        a = x1 + x2
+    }
+}
+
+class Bar(val a: Int) {
+    constructor(b: Boolean) : this(when (b) {
+                                        true -> 1
+                                        false -> 0
+                                    }) {
```
Comment (22slin22) [inline plugins/formal-verification/testData/diagnostics/no_contracts/class_constructors.kt:22]: Nit: The formatting here seems a little off. I that what the kotlin linter recommends?

Comment (gabryon99) [reply plugins/formal-verification/testData/diagnostics/no_contracts/class_constructors.kt:22]: My IntelliJ doesn't run the linter on the Kotlin test files 🤔
I can try to simplify it with an `if` statement!

File: plugins/formal-verification/testData/diagnostics/no_contracts/classes_setters.fir.txt
Comment (22slin22) [inline plugins/formal-verification/testData/diagnostics/no_contracts/classes_setters.fir.txt:1]: You forgot to remove this FIR file

**Review by 22slin22 — APPROVED** [review]

**Pull request merged**